tools/downloader/topologies_test.py

```diff
@@ -0,0 +1,103 @@
+import unittest
+import os
+import numpy as np
+import cv2 as cv
+
+import topologies as omz
+
+from pathlib import Path
+from parameterized import parameterized
+from openvino import inference_engine
+from openvino.inference_engine import IECore
+
+class TestTopologies(unittest.TestCase):
+    @parameterized.expand(['FP16', 'FP32'])
+    def test_load_intel(self, p):
+        face_detection = omz.Model('face-detection-0100')
+        xml_path, bin_path = face_detection.get_ir(precision=p)
+        
+        self.assertTrue(os.path.exists(xml_path))
+        self.assertTrue(os.path.exists(bin_path))
+
+        self.assertEqual(os.path.basename(xml_path), 'face-detection-0100.xml')
+        self.assertEqual(os.path.basename(bin_path), 'face-detection-0100.bin')
+
+        self.assertEqual(os.path.basename(os.path.dirname(xml_path)), p)
+        self.assertEqual(os.path.basename(os.path.dirname(bin_path)), p)
```

Comment:
I don't think the conditions verified by these assertions are part of the library's contract. The documentation just says that the `get_ir` will give you a path to the model files; it doesn't say what the path will _be_. What you should be checking is that a) the model can be loaded into Inference Engine, and b) that it actually has the requested precision. (To be fair, I'm not sure if the latter _can_ be checked. But if you can do it, you should.)

Reply:
I can't find a way to check model precision.